docs/usage/configuration-options.md

```diff
@@ -1568,6 +1568,157 @@ Example:
 }
 ```
 
+### matchStringsStrategy
+
+`matchStringsStrategy` defines which procedure should be used to interpret the with `matchString` defined regular expressions.
+Three options are available:
+
+- `any` (default)
+- `recursive`
+- `combination`
+
+#### any
+
+Each provided `matchString` will be matched individually to the content of the `packageFile`.
+If a `matchString` has multiple matches in a file each will be interpreted as an independent dependency.
+
+As example the following configuration will update all 3 lines in the Dockerfile.
+renovate.json:
+
+```json
+{
+  "regexManagers": [
+    {
+      "fileMatch": ["^Dockerfile$"],
+      "matchStringsStrategy": "any",
+      "matchStrings": [
+        "ENV [A-Z]+_VERSION=(?<currentValue>.*) # (?<datasource>.*?)/(?<depName>.*?)(\\&versioning=(?<versioning>.*?))?\\s",
+        "FROM (?<depName>\\S*):(?<currentValue>\\S*)"
+      ],
+      "datasourceTemplate": "docker"
+    }
+  ]
+}
+```
+
+a Dockerfile:
+
+```dockerfile
+FROM amd64/ubuntu:18.04
+
+ENV GRADLE_VERSION=6.2 # gradle-version/gradle&versioning=maven
+ENV NODE_VERSION=10.19.0 # github-tags/nodejs/node&versioning=node
+```
+
+#### recursive
+
+If using `recursive` the `matchString` will be looped through and the full match of the last will define the range of the next one.
+This can be used to narrow down the search area to prevent multiple matches.
+How ever the `recursive` strategy allows to handle multiple dependencies. The workflow is as follows.
+All matches of the first `matchString` are detected, for each of these matches the match range will used as basis for the second `matchString`.
+If this `matchString` again has multiple matches it will split again. This process will be followed as long there is a match,
+a regex is available or, a dependency is detected.
```

Comment:
```suggestion
If the next `matchStrings` pattern has multiple matches then it will split again.
This process will be followed as long there is a match plus a next `matchingStrings` pattern is available or a dependency is detected.
```

I don't get the last bit though. Does it stop using `matchStrings` if a dependency is already detected?